File: back-end/src/main/java/app/web/gprojuridico/model/Assistido.java
package app.web.gprojuridico.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@ToString
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Assistido {
    private String nome;
    private String email;
    private String cpf;
    private String rg;
    private String naturalidade;
    private String nacionalidade;
    private String dataNascimento;
    private String estadoCivil;
    private String telefone;
    private String cidade;
    private String cep;
    private String enderecoResidencial;
    private String escolaridade;
    private String nomePai;
    private String nomeMae;
    private String profissao;
    private String remuneracao;
    private String cidadeComercial;
    private String enderecoComercial;
    private String numDependentes;
    private String documentId;

}

File: src/app/core/types/assistido.ts
import { Endereco } from './endereco';

export interface Assistido {
  '@type'?: string,
  nome: string,
  email?: string,
  cpf: string,
  rg: string,
  nacionalidade?: string,
  estadoCivil?: string,
  telefone?: string,
  endereco: TipoEndereco,
  escolaridade?: string,
  filiacao: Filiacao,
  profissao?: string,
  remuneracao?: string,
}

export interface AssistidoCivil extends Assistido {
  dataNascimento?: string,
  naturalidade?: string,
  dependentes?: number,
}

export interface AssistidoTrabalhista extends Assistido {
  ctps: Ctps,
  pis: string,
  empregadoAtualmente: boolean,
}

export interface AssistidoFull extends AssistidoCivil, AssistidoTrabalhista { }

export interface Filiacao {
  pai?: string,
  mae: string,
}

export interface TipoEndereco {
  residencial: Endereco,
  comercial?: Endereco,
}

export interface Ctps {
  numero?: string,
  serie?: string,
  uf?: string,
}

